e2e/src/invite-to-pay/agent.spec.ts

```diff
@@ -0,0 +1,216 @@
+import { test, expect } from "@playwright/test";
+import {
+  setFeatureFlag,
+  addSessionToContext,
+  cards,
+  fillGovUkCardDetails,
+  modifyFlow,
+} from "../helpers";
+import inviteToPayFlow from "../flows/invite-to-pay-flow";
+import {
+  Context,
+  getGraphQLClient,
+  setUpTestContext,
+  tearDownTestContext,
+} from "../context";
+import {
+  answerInviteToPayForm,
+  getPaymentRequestBySessionId,
+  makePaymentRequest,
+  navigateToPayComponent,
+} from "./helpers";
+import { mockPaymentRequest, modifiedInviteToPayFlow } from "./mocks";
+
+let context: Context = {
+  user: {
+    firstName: "test",
+    lastName: "test",
+    email: "e2epaytest@opensystemslab.com",
+  },
+  team: {
+    name: "Buckinghamshire",
+    slug: "buckinghamshire",
+    logo: "https://placedog.net/250/250",
+    primaryColor: "#F30415",
+    homepage: "example.com",
+  },
+  flow: {
+    slug: "invite-to-pay-test",
+    data: inviteToPayFlow,
+  },
+  sessionIds: [], // used to collect and clean up sessions
+};
+
+test.describe("Agent journey", async () => {
+  const adminGQLClient = getGraphQLClient();
+
+  test.beforeAll(async () => {
+    try {
+      context = await setUpTestContext(context);
+    } catch (e) {
+      // ensure proper teardown if setup fails
+      await tearDownTestContext(context);
+      throw e;
+    }
+  });
+
+  test.beforeEach(
+    async ({ page }) => await setFeatureFlag(page, "INVITE_TO_PAY")
+  );
+
+  test.afterAll(async () => await tearDownTestContext(context));
+
+  test("agent can send a payment request", async ({ page }) => {
+    await navigateToPayComponent(page, context);
+    const sessionId = await addSessionToContext(page, context);
+
+    const toggleInviteToPayButton = page.getByRole("button", {
+      name: "Invite someone else to pay for this application",
+    });
+    expect(toggleInviteToPayButton).toBeVisible();
+    await toggleInviteToPayButton.click();
+    const inviteToPayFormHeader = await page.getByText(
+      "Invite someone else to pay for this application"
+    );
+    expect(inviteToPayFormHeader).toBeVisible();
+
+    await answerInviteToPayForm(page);
+    await page.getByText("Send invitation to pay").click();
+    await page.waitForLoadState("networkidle");
```

Comment:
nit: It's worth noting that waiting for `networkidle` is discouraged in the Playwright docs: https://playwright.dev/docs/api/class-page#page-wait-for-load-state
I don't think this needs to change now but we should review an approach to notifying event completion (listening for debug logs isn't ideal either but that's my best stab at this so far).